File: api/src/main/java/gg/mineral/api/entity/meta/EntityMetadata.java
package gg.mineral.api.entity.meta;

import lombok.RequiredArgsConstructor;

public interface EntityMetadata {

    @RequiredArgsConstructor
    public static final class Entry implements Comparable<Entry> {
        public final EntityMetadataIndex index;
        public final Object value;

        @Override
        public int compareTo(Entry o) {
            return o.index.getIndex() - index.getIndex();
        }

        @Override
        public String toString() {
            return index + "=" + value;
        }
    }
}

File: api/src/main/kotlin/gg/mineral/api/entity/meta/EntityMetadata.kt
package gg.mineral.api.entity.meta

interface EntityMetadata {
    class Entry(val index: EntityMetadataIndex, val value: Any) : Comparable<Entry> {
        override fun compareTo(other: Entry): Int = other.index.index - index.index

        override fun toString(): String = "$index=$value"
    }
}
